src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,154 @@
+                } catch (DateTimeParseException e) {
+                    ui.exceptionMessage(new DukeException(
+                            "☹ OOPS!!! Please provide a valid date with the format yyyy-mm-dd."));
+                } catch (Exception e) {
+                    ui.exceptionMessage(new DukeException(
+                            "☹ OOPS!!! Please provide a date using '/by ' with the format yyyy-mm-dd."));
+                }
+            } catch (IndexOutOfBoundsException e) {
+                ui.exceptionMessage(new DukeException("☹ OOPS!!! The description of a deadline cannot be empty."));
+            }
+            break;
+        case "event":
+            try {
+                String[] taskDetails = inputs[1].split("/at ");
+                if (taskDetails[0].trim().isEmpty()) {
+                    throw new IndexOutOfBoundsException();
+                }
+                try {
+                    LocalDate date = LocalDate.parse(taskDetails[1]);
+                    Task event = new Event(taskDetails[0].trim(), date);
+                    tasks.add(event);
+                    data = "E | 0 | " + taskDetails[0].trim() + " | "
+                            + date.format(DateTimeFormatter.ofPattern("MMM d yyyy")) + "\n";
+                    storage.writeToFile(data);
+                    ui.userMessage("Got it. I've added this task:\n    "
+                            + event + "\nNow you have " + tasks.size() + " tasks in the list.");
+                } catch (DateTimeParseException e) {
+                    ui.exceptionMessage(new DukeException(
+                            "☹ OOPS!!! Please provide a valid date with the format yyyy-mm-dd."));
+                } catch (Exception e) {
+                    ui.exceptionMessage(new DukeException(
+                            "☹ OOPS!!! Please provide a date using '/at ' with the format yyyy-mm-dd.."));
+                }
+            } catch (IndexOutOfBoundsException e) {
+                ui.exceptionMessage(new DukeException("☹ OOPS!!! The description of a event cannot be empty."));
+            }
+            break;
+        case "find":
+            try {
+                ArrayList<Task> result = tasks.find(inputs[1]);
+                if (result.size() == 0) {
+                    ui.userMessage("No match found.");
+                    break;
+                }
+                ui.userMessage("Here are the matching tasks in your list:");
+                for (int i = 0; i < result.size(); i++) {
+                    ui.userMessage(i + 1 + "." + result.get(i));
+                }
+            } catch (IndexOutOfBoundsException e) {
+                ui.exceptionMessage(new DukeException("☹ OOPS!!! The keyword for find cannot be empty."));
+            }
+            break;
+        default:
+            ui.exceptionMessage(new DukeException("☹ OOPS!!! I'm sorry, but I don't know what that means :-("));
+            break;
+        }
+        return true;
+    }
+}
```

Comment:
Your Parser class seems like it's doing a lot of work. Maybe you could refract it by adding more classes such as AddCommand, DeleteCommand and DoneCommand to lighten the workload of your Parser.